clients/java/client/src/it/java/org/camunda/bpm/client/variable/FileSerializationIT.java

```diff
@@ -163,6 +204,48 @@ public void shouldGetTyped_Deferred() {
     assertThat(typedValue.isLoaded()).isFalse();
     assertThat(typedValue.getEncoding()).isNull();
     assertThat(typedValue.getMimeType()).isNull();
+
+    DeferredFileValueImpl typedValueImpl = (DeferredFileValueImpl) typedValue;
+    assertThat(typedValueImpl.getExecutionId()).isEqualTo(task.getExecutionId());
+  }
+
+  @Test
+  public void shouldGetVariableTypedForLocalVariable() {
+    // given
+    ProcessDefinitionDto processDefinitionDto = engineRule.deploy(
+        Bpmn.createExecutableProcess("process")
+        .startEvent("startEvent")
+        .serviceTask("serviceTaskFoo")
+          .camundaExternalTask(EXTERNAL_TASK_TOPIC_FOO)
+            // create the local file variable with the same content but different name
+          .camundaInputParameter(LOCAL_VARIABLE_NAME_FILE, "${execution.getVariableTyped('fileVariable')}")
+        .serviceTask("serviceTaskBar")
+          .camundaExternalTask(EXTERNAL_TASK_TOPIC_BAR)
+        .endEvent("endEvent")
+        .done()
+    ).get(0);
+
+    engineRule.startProcessInstance(processDefinitionDto.getId(), VARIABLE_NAME_FILE, VARIABLE_VALUE_FILE);
+
+    // when
+    client.subscribe(EXTERNAL_TASK_TOPIC_FOO)
+        .handler(handler)
+        .open();
+
+    clientRule.waitForFetchAndLockUntil(() -> !handler.getHandledTasks().isEmpty());
+
+    ExternalTask task = handler.getHandledTasks().get(0);
+
+    // then
+    DeferredFileValue typedValue = task.getVariableTyped(LOCAL_VARIABLE_NAME_FILE);
+    assertThat(typedValue.getFilename()).isEqualTo(VARIABLE_VALUE_FILE_NAME);
+    assertThat(typedValue.getType()).isEqualTo(FILE);
+    assertThat(typedValue.isLoaded()).isFalse();
+    assertThat(typedValue.getEncoding()).isNull();
+    assertThat(typedValue.getMimeType()).isNull();
+
+    DeferredFileValueImpl typedValueImpl = (DeferredFileValueImpl) typedValue;
+    assertThat(typedValueImpl.getExecutionId()).isEqualTo(task.getExecutionId());
```

Comment:
🔧 Let's keep using a white box test approach to a minimum. A black box should be the way to go.
We can fall back on white box tests only when a black box test is too difficult to achieve.